go/pkg/command/command.go

```diff
@@ -0,0 +1,273 @@
+// Package command defines common types to be used with remote execution.
+package command
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"sort"
+	"time"
+
+	"github.com/bazelbuild/remote-apis-sdks/go/digest"
+	"github.com/pborman/uuid"
+)
+
+// InputKind can be specified to narrow down the matching for a given input path.
+type InputKind int
+
+const (
+	// Any input type will match.
+	UnspecifiedInputKind InputKind = iota
+
+	// Only directories match.
+	DirectoryInputKind
+
+	// Only files match.
+	FileInputKind
+)
+
+func (s InputKind) String() string {
+	return [...]string{"UnspecifiedInputKind", "DirectoryInputKind", "FileInputKind"}[s]
+}
+
+// InputExclusion represents inputs to be excluded from being uploaded for remote execution.
+type InputExclusion struct {
+	// Required: the path regular expression to match for exclusion.
+	Regex string
+
+	// The input kind to match for exclusion.
+	Kind InputKind
+}
+
+// InputSpec represents all the required inputs to a remote command.
+type InputSpec struct {
+	// Input paths (files or directories) that need to be uploaded to the remote
+	// server for the command execution.
+	// Note: a known limitation of the first version of the proxy is that for any
```

Comment:
Same comment about uploading. Also the proxy is irrelevant here right?

Reply:
Ah, derp, that's a copy-paste bug!